docs/contributing/index.rst

```diff
@@ -4,200 +4,464 @@
 
-  To avoid duplicating work, it is highly advised that you search through the
-  `issue tracker <https://github.com/geomstats/geomstats/issues>`_ and
-  the `PR list <https://github.com/geomstats/geomstats/pulls>`_.
-  If in doubt about duplicated work, or if you want to work on a non-trivial
-  feature, it's recommended to first open an issue in
-  the `issue tracker <https://github.com/geomstats/geomstats/issues>`_
-  to get some feedbacks from core developers.
+#. | At this point you should have the geomstats code on your machine ready for development. Create a new development branch for further development:
 
+    .. code-block:: bash
 
-Contributing: Adding a new geometry or manifold
------------------------------------------------
+      $ git checkout -b feature-branch
+   
 
-  Get used to the code design of geomstats' classes by having a look at the `template manifold <https://github.com/geomstats/geomstats/tree/master/geomstats/geometry/_my_manifold.py>`__ and `template unit-tests <https://github.com/geomstats/geomstats/blob/master/tests/tests_geomstats/test__my_manifold.py>`__ implementations.
+7. Verify that you are on the new branch:
 
-Contributing: GitHub Workflow
-=============================
+    .. code-block:: bash
 
-The preferred way to contribute to geomstats is to fork the `main
-repository <https://github.com/geomstats/geomstats/>`__ on GitHub,
-then submit a "pull request" (PR).
+      $ git branch
+      * feature-branch
+      master
+
+8. Verify your remote configuration:
+
+    .. code-block:: bash
+
+      $ git remote -v
+      upstream	https://github.com/geomstats/geomstats (fetch)
+      upstream	https://github.com/geomstats/geomstats (push)
+      origin	https://github.com/<username>/geomstats (fetch)
+      origin	https://github.com/<username>/geomstats (push)
+
+.. _dependencies
+
+Dependencies and a virtual environment
+---------------------------------------
+
+We recommend using the conda virtual environment to create separation between your development environment
+from any other geomstats versions installed on your system. 
+
+Create a virtual environment with conda using:
+
+    .. code-block:: bash
+
+      $ conda create -n geomstats-dev
+
+Then clone your code to this environment and activate it:
+
+    .. code-block:: bash
+
+      $ git clone git@github.com:<username>/geomstats.git
+      $ cd geomstats
+      $ conda activate geomstats-dev
+
+Now install the geomstats minimum requirements:
+
+    .. code-block:: bash
+
+      $ conda env create --file environment.yml
+
+.. note::
+   See the `setup.cfg` file for details on all project requirements.
+
+.. _backends dev
+
+Backends
+--------
+
+Geomstats supports several backends, namely:
+
+- numpy
+- autograd
+- tensorflow
+- pytorch
+
+The default backend is `numpy`, install the other backends using:
+
+    .. code-block:: bash  
+
+      $ pip install geomstats[<backend_name>]
```

Comment:
maybe add an example (just so people do not literally write <backend_name> (I know I could have several years ago)) but this is really unimportant

Reply:
I think its common convention to use fillers like these especially if we have many option. Inclined to keep as is.